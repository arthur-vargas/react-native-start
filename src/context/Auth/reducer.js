export const reducer = (state, action) => {
  switch (action.type) {
    case 'add_error':
      return { ...state, errorMessage: action.payload };
    case 'signin':
      return { errorMessage: '', token: action.payload };
    case 'clear_error_message':
      return { ...state, errorMessage: '' };
    case 'signout':
      return { token: null, errorMessage: '' };
    default:
      return state;
  }
};
