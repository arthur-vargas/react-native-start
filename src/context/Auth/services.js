import AsyncStorage from '@react-native-community/async-storage';
import authApi from '../../api/auth';
import { navigate } from '../../navigationRef';

export const tryLocalSignin = (dispatch) => async () => {
  const token = await AsyncStorage.getItem('token');
  if (token) {
    dispatch({ type: 'signin', payload: token });
    navigate('Home');
  } else {
    navigate('Login');
  }
};

export const clearErrorMessage = (dispatch) => () => {
  dispatch({ type: 'clear_error_message' });
};

export const signin = (dispatch) => async ({ username, password }) => {
  try {
    const response = await authApi.post('/users/authenticate', { username, password });
    await AsyncStorage.setItem('token', response.data.token);
    dispatch({ type: 'signin', payload: response.data.token });
    navigate('Home');
  } catch (err) {
    console.log(err); // eslint-disable-line no-console
    dispatch({
      type: 'add_error',
      payload: 'Something went wrong with sign in',
    });
  }
};

export const signout = (dispatch) => async () => {
  await AsyncStorage.removeItem('token');
  dispatch({ type: 'signout' });
  navigate('Login');
};
