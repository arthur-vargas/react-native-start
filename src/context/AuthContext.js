import AsyncStorage from '@react-native-community/async-storage';
import createDataContext from './createDataContext';
import authApi from '../api/auth';
import { navigate } from '../navigationRef';

const authReducer = (state, action) => {
  switch (action.type) {
    case 'add_error':
      return { ...state, errorMessage: action.payload };
    case 'signin':
      return { errorMessage: '', token: action.payload };
    case 'clear_error_message':
      return { ...state, errorMessage: '' };
    case 'signout':
      return { token: null, errorMessage: '' };
    default:
      return state;
  }
};

const tryLocalSignin = (dispatch) => async () => {
  const token = await AsyncStorage.getItem('token');
  if (token) {
    dispatch({ type: 'signin', payload: token });
    navigate('Home');
  } else {
    navigate('Login');
  }
};

const clearErrorMessage = (dispatch) => () => {
  dispatch({ type: 'clear_error_message' });
};

const signin = (dispatch) => async ({ username, password }) => {
  try {
    const response = await authApi.post('/users/authenticate', { username, password });
    await AsyncStorage.setItem('token', response.data.token);
    dispatch({ type: 'signin', payload: response.data.token });
    navigate('Home');
  } catch (err) {
    console.log(err); // eslint-disable-line no-console
    dispatch({
      type: 'add_error',
      payload: 'Something went wrong with sign in',
    });
  }
};

const signout = (dispatch) => async () => {
  await AsyncStorage.removeItem('token');
  dispatch({ type: 'signout' });
  navigate('Login');
};

export const { Provider, Context } = createDataContext(
  authReducer,
  {
    signin,
    signout,
    clearErrorMessage,
    tryLocalSignin,
  },
  {
    token: null,
    errorMessage: '',
  },
);
