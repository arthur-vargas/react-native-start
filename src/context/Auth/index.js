import createDataContext from '../createDataContext';
import * as actions from './services';
import { reducer } from './reducer';

export const { Provider, Context } = createDataContext(
  reducer,
  {
    ...actions,
  },
  {
    token: null,
    errorMessage: '',
  },
);
