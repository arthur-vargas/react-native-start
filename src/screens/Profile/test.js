/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import renderer from 'react-test-renderer';
import firebase from 'react-native-firebase';
import { shallow } from 'enzyme';

import Screen from './index';
import { Provider as AuthProvider } from '../../context/Auth';

describe('Screen: Profile', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
    services: {
      firebase,
    },
  };

  beforeEach(() => {
  });

  it('renders correctly', () => {
    renderer.create(<AuthProvider><Screen {...props} /></AuthProvider>);
  });

  it('Render', () => {
    expect(
      shallow(<AuthProvider><Screen {...props} /></AuthProvider>),
    ).toMatchSnapshot();
  });
});
