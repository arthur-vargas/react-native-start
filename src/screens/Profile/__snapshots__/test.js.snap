// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: Profile Render 1`] = `ShallowWrapper {}`;
